### PR TITLE
Fix label corsika histograms

### LINKED ISSUE #637: fix label in CORSIKA histogram
The histogram saved as `hist_1D_photon_density_distr_all_tels` by `generate_corsika_histograms.py` provides `counts` in the y label, while it is supposed to be unit of 1/Area.

### PR DESCRIPTION
**Pull request by VictorBarbosaMartins**
Closes #637.

Fixed the y label of the 1D density histogram, and also the y label of other 2D histograms which were appearing as units "()". We should come up with nicer names for the histograms instead of writing the name of the module function as the title of the histogram. It is probably better to do that in another PR.

Thanks.

### CODE REVIEW

File: simtools/corsika/corsika_histograms_visualize.py [hunk 1]
```diff
@@ -268,7 +268,13 @@ def _kernel_plot_1D_photons(histograms_instance, property_name, log_y=True):
             )
         else:
             ax.set_xlabel(f"{histograms_instance._dict_1D_distributions[property_name]['edges']} ")
-        ax.set_ylabel("Counts")
+        if property_name == "density":
+            ax.set_ylabel(
+                f"Density ("
+                f"{histograms_instance._dict_1D_distributions[property_name]['edges unit']}$^{-2}$)"
```
Comment (GernotMaier) [inline simtools/corsika/corsika_histograms_visualize.py:274]: I think the superscript did not work:

Screenshot 2023-10-16 at 14 46 48

**Review by GernotMaier — CHANGES_REQUESTED** [review]
see comment on the superscript.

File: simtools/utils/general.py [hunk 2]
```diff
@@ -858,9 +858,9 @@ def convert_2D_to_radial_distr(hist2d, xaxis, yaxis, bins=50, max_dist=1000):
     hist2d: numpy.ndarray
         The histogram counts.
     xaxis: numpy.array
-        The values of the x axis (histogram edges) on the ground.
+        The values of the x axis (histogram bin_edges) on the ground.
```
Comment (GernotMaier) [inline simtools/utils/general.py:861]: I think in the comment you don't need the underscore in `bin edges`

File: simtools/utils/general.py [hunk 3]
```diff
@@ -858,9 +858,9 @@ def convert_2D_to_radial_distr(hist2d, xaxis, yaxis, bins=50, max_dist=1000):
...
     yaxis: numpy.array
-        The values of the y axis (histogram edges) on the ground.
+        The values of the y axis (histogram bin_edges) on the ground.
```
Comment (GernotMaier) [inline simtools/utils/general.py:863]: Same as above

File: simtools/utils/general.py [hunk 4]
```diff
@@ -871,7 +871,7 @@ def convert_2D_to_radial_distr(hist2d, xaxis, yaxis, bins=50, max_dist=1000):
     np.array
         The values of the 1D histogram with size = int(max_dist/bin_size).
     np.array
-        The edges of the 1D histogram with size = int(max_dist/bin_size) + 1.
+        The bin_edges of the 1D histogram with size = int(max_dist/bin_size) + 1.
```
Comment (GernotMaier) [inline simtools/utils/general.py:874]: Same as above

File: tests/unit_tests/corsika/test_corsika_histograms.py [hunk 5]
```diff
@@ -318,7 +322,7 @@ def test_get_2D_photon_position_distr(corsika_histograms_instance_set_histograms
     counts = corsika_histograms_instance_set_histograms.get_2D_photon_position_distr()
     assert pytest.approx(np.sum(counts[0]), 1e-2) == 11633
 
-    # The edges should be the same
+    # The bin_edges should be the same
```
Comment (GernotMaier) [inline tests/unit_tests/corsika/test_corsika_histograms.py:325]: same as above

File: tests/unit_tests/corsika/test_corsika_histograms.py [hunk 6]
```diff
@@ -355,7 +359,7 @@ def test_get_2D_num_photons_distr(corsika_histograms_instance_set_histograms):
         telescope_indices_array,
     ) = corsika_histograms_instance_set_histograms.get_2D_num_photons_distr()
     assert np.shape(num_events_array) == (1, 3)  # number of events in this output file + 1
-    # (edges of hist)
+    # (bin_edges of hist)
```
Comment (GernotMaier) [inline tests/unit_tests/corsika/test_corsika_histograms.py:362]: same as above

**Review by GernotMaier — APPROVED** [review]
Very minor comment, but all good!

**Comment by VictorBarbosaMartins** [comment]
Thanks for the remaining comments and the approval. I implemented them and the tests are passing, so I am merging the PR.

**Pull request merged**